week2/intro_to_property_packages_verbose.ipynb

```diff
@@ -0,0 +1,681 @@
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "\n",
+    "from watertap.property_models.seawater_prop_pack import SeawaterParameterBlock\n",
+    "from watertap.property_models.NaCl_prop_pack import NaClParameterBlock\n",
+    "from watertap.property_models.NaCl_T_dep_prop_pack import (\n",
+    "    NaClParameterBlock as NaClParameterBlock_Tdep,\n",
+    ")\n",
+    "from watertap.core.solvers import get_solver"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0676ed18",
+   "metadata": {},
+   "source": [
+    "### Create model, flowsheet, and property model\n",
+    "\n",
+    "Like any WaterTAP model, we first create the model, flowsheet, and property model. \n",
+    "\n",
+    "<center><img src=\"graphics/prop_pack1.png\" width=\"400\" /></center>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8ca5373d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = ConcreteModel()\n",
+    "m.fs = FlowsheetBlock(dynamic=False)\n",
+    "m.fs.properties = SeawaterParameterBlock()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89bd5989",
+   "metadata": {},
+   "source": [
+    "### Add `Feed` model to flowsheet\n",
+    "\n",
+    "Next, we add the feed model to the flowsheet and pass the property model we created previously via the `property_package` argument. \n",
+    "\n",
+    "As soon as this cell is run, our `feed` model is built. As part of the build process, a _state block_ is created. Despite being termed a \"property model\", the `m.fs.properties` block contains no stream properties on it. Instead, it contains all the information needed to create properties on _state blocks_. Stream properties are tracked from unit model to unit model along the flowsheet.\n",
+    "\n",
+    "For the `Feed` model, the _state block_ is called `properties` and is indexed to `[0]`. \n",
+    "\n",
+    "<center><img src=\"graphics/prop_pack2.png\" width=\"400\"/></center>\n",
+    "\n",
+    "Like many other components, we can use the `display()` method to see all the variables, constraints, and objectives on the state block.\n",
+    "\n",
+    "In this case, we observe only the _state variables_. These are the specific variables used by the property model to define the state of the stream. \n",
+    "\n",
+    "For the seawater property model (and many other WaterTAP property models), the state variables are:\n",
+    "- pressure\n",
+    "- temperature\n",
+    "- mass flow rate of components \n",
+    "\n",
+    "For the seawater property model, the components are `H2O` and `TDS`, but other property models allow adding any number of components."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a32b0bc1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.fs.feed = Feed(property_package=m.fs.properties)\n",
+    "m.fs.feed.report()\n",
+    "# m.fs.feed.properties[0].display()"
```

Comment:
unclear if you intentionally commented out or did this temporarily. I think it can stay commented out anyhow